docs/madr/decisions/033-meshhttproute-meshgateway.md

```diff
@@ -0,0 +1,104 @@
+# `MeshHTTPRoute` for `MeshGateway`
+
+- Status: accepted
+
+Technical Story: #8142
+
+## Context and Problem Statement
+
+We have two very similar resources, `MeshHTTPRoute` and `MeshGatewayRoute` that
+both configure route matching, filtering and redirecting semantics.
+This MADR proposes adapting `MeshHTTPRoute` for use with
+`MeshGateway.`
+
+## Decision Drivers <!-- optional -->
+
+* Fewer CRDs
+* Parity with `MeshGatewayRoute`
+
+## Considered Options
+
+- Keep two resources
+- Use `MeshHTTPRoute`
+
+## Decision Outcome
+
+Chosen option:
+
+- Adapt `MeshHTTPRoute`
+- Add `spec.to[].hostnames`
+- Allow `spec.to[].targetRef` to be empty
+- Require `backendRefs` to be set
+- Don't support policy matching on `MeshHTTPRoute` if it attaches to a gateway
+
+### Hostname
+
+With `MeshGatewayRoute` and `HTTPRoute`, users can set hostnames to only apply
+the route rules to requests going to the given hosts.
+We don't have this possibility with `MeshHTTPRoute`, so we can either:
+
+- add `hostnames` to the `MeshHTTPRoute`, likely directly under `spec.to`
+  entries
+- have users on each rule do hostname matching using the header match on `host`
+
+This MADR proposes adding `spec.to[].hostnames`. Hostname matching is in some sense the
+"highest level" matching possible with HTTP and it's likely that all rules of a
+given resource are intended to apply to a given hostname set, whether that's a
+specific hostname, a wildcard pattern or to all hostnames.
+
+The second option might be feasible but it only makes sense if the cost of having
+host matching separate is very high. It also differs from Gateway API, which is
+a disadvantage.
+
+### `to.targetRef`
+
+When using `MeshHTTPRoute` for service to service communication, it's very
+useful to limit the effects based on request source and destination. We do
+this by setting `spec.targetRef`, the source, and `spec.to[].targetRef`, the destination.
+
+With `MeshGateway`, once a request hits the gateway, the routing resources
+_define_ the destination. We have to put the rules somewhere, so this MADR
+proposes `spec.to` but there's no semantic value to a `spec.to[].targetRef`
+or for that matter, a `spec.from[].targetRef`.
```

Comment:
So to be clear here is that `spec.to[].target` would be unset if the top level targetRef is MeshGateway?
Again an example would be good.
Would it be not allowed to bind a route using MeshSubset or MeshService? .i.e if the DPP is a gateway then we only look at MeshGateway top level targetRefs? Does that make policy matching quite different for routes?

Reply:
Yes it would be not allowed. I guess it's different but the intention AFAIU at the moment anyway is to avoid MeshService and friends for matching on gateways.

Reply:
Yeah this is already present in the example

Reply:
@lahabana WDYT? The problem with "allowing" `MeshService` and friends for gateways is that it prevents validation. In particular, atm we _disallow_ `kind: Mesh` for non-MeshGateway MeshHTTPRoutes.

Reply:
I'm not sure I understand the precise question.
But I'm fine with having only MeshGateway work for gateways.

Reply:
> But I'm fine with having only MeshGateway work for gateways 

That was the question. Yeah we should probably ignore such routes.

Reply:
We will have to ignore such routes in some cases, like `from` with `MeshFaultInjection`